test/config_test/BUILD

```diff
@@ -41,17 +80,15 @@ envoy_cc_test_library(
         "//source/server/config_validation:server_lib",
         "//test/integration:integration_lib",
         "//test/mocks/server:instance_mocks",
-        "//test/mocks/server:worker_factory_mocks",
         "//test/mocks/server:listener_component_factory_mocks",
+        "//test/mocks/server:worker_factory_mocks",
         "//test/mocks/server:worker_mocks",
         "//test/mocks/ssl:ssl_mocks",
-        "//test/test_common:threadsafe_singleton_injector_lib",
         "//test/test_common:simulated_time_system_lib",
-    ] + select({
-        "//bazel:windows_x86_64": envoy_all_extensions(WINDOWS_SKIP_TARGETS),
-        "//bazel:linux_ppc": envoy_all_extensions(PPC_SKIP_TARGETS),
-        "//conditions:default": envoy_all_extensions(),
```

Comment:
@htuch May I ask you what would be a better solution here? The problem is that the tests `//test/config_test/...` fail when the brotli (de)compressor is added to `envoy_all_extensions()` and ASan instrumentalization is added to the test binaries (presumably due to excessive size of the code segments). I kinda solved the problem by linking `//test/config_test:example_configs_test` with those extensions only that are used in the example configs. And `//test/config_test:deprecated_configs_test` doesn't seem to depend on `envoy_all_extensions()` at all.

But I suspect that the tool `//test/tools/config_load_check:config_load_check_tool` is useless now. I tried to link the tool with `envoy_all_extensions()`, but stumbled upon visibility constraints:
```
ERROR: /source/test/tools/config_load_check/BUILD:14:22: in cc_library rule //test/tools/config_load_check:config_load_check_lib: target '//source/extensions/internal_redirect/allow_listed_routes:config' is not visible from target '//test/tools/config_load_check:config_load_check_lib'. Check the visibility declaration of the former target if you think the dependency is legitimate
```